In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
%cd /content/drive/MyDrive/boaz/

/content/drive/MyDrive/boaz


In [ ]:
import pandas as pd
df1 = pd.read_csv('credit_card_data.csv')
df2 = pd.read_csv('jeju_financial_life_data.csv')
pd.set_option('display.max_columns', None)

# 1. credit_card_data.csv 
> __연령대/거주지/성별__ 구분에 따른 __카드 및 대출__ 이용통계 데이터로, 구분 값 별 카드이용금액 비교, 대출금액 및 대출 특이사항 비교 등이 가능하다.

## __전국민 카드 및 대출 이용통계 데이터__

### 기본정보
1. 'pop_cd': 고객구분 코드
2. 'year': 연도
3. 'month': 월
4. 'city': 지역
5. 'sex': 성별
6. 'ages': 연령대
7. 'population': 해당 그룹 고객 인원 수

### 신용정보
1. 'avg_score': 고객구분 코드 그룹의 평균 평점
2. 'avg_rat': 고객구분 코드 그룹의 평균 등급


### 카드정보
1. 'num_opencard': 소지 카드 개수 
2. 'num_usecard': 실제 사용 카드 개수

### 월별정보
1. 'monthly_card_spend': 월별 카드 소비액
2. 'monthly_lc'
3. 'monthly_loan': 월별 대출 
4. 'monthly_bk_loan'
5. 'monthly_cd_loan': 월별 카드 대출 
6. 'monthly_installments_loan': 월별 할부 대출
7. 'monthly_insurance_loan': 월별 보험 대출 
8. 'monthly_sbk_loan'

### 대출정보
1. 'loan_commitment': 대출 약정 금액
2. 'inst_rep_loanb'
3. 'ls_rep_loanb'
4. 'credit_loan': 카드 대출
5. 'mortgage_loan': 주택 대출
6. 'credit_card_payment': 카드 지출
7. 'credit_card_installments_payment': 카드 할부 지출

In [ ]:
df1

pop_cd  year  month  avg_score  avg_rat city  sex ages  population  \
0      L011  2016      1        812        2   서울  NaN  10대       55000   
1      L011  2016      2        773        4   서울  NaN  10대      380000   
2      L011  2016      3        771        4   서울  NaN  10대      387000   
3      L011  2016      4        770        4   서울  NaN  10대      412000   
4      L011  2016      5        771        4   서울  NaN  10대      414000   
...     ...   ...    ...        ...      ...  ...  ...  ...         ...   
3883    Y95  2017      8        819        3  NaN   여자  90대      232000   
3884    Y95  2017      9        819        3  NaN   여자  90대      235000   
3885    Y95  2017     10        819        3  NaN   여자  90대      239000   
3886    Y95  2017     11        819        3  NaN   여자  90대      243000   
3887    Y95  2017     12        820        3  NaN   여자  90대      265000   

      num_opencard  num_usecard  monthly_card_spend  monthly_lc  monthly_loan  \
0             1.44         1.13               70000    92630000      87350000   
1             1.71         0.96              170000    31140000      28910000   
2             1.75         0.95              190000    30230000      28180000   
3             1.76         0.92              200000    30430000      27710000   
4             1.75         0.94              170000    31210000      27620000   
...            ...          ...                 ...         ...           ...   
3883          0.11         0.03               90000   100520000      86580000   
3884          0.11         0.03               90000   100580000      86780000   
3885          0.11         0.03               90000   100380000      86110000   
3886          0.11         0.03               90000   100370000      85920000   
3887          0.11         0.03               90000   101890000      86660000   

      monthly_bk_loan  monthly_cd_loan  monthly_installments_loan  \
0            85790000                0                          0   
1            25850000           190000                     510000   
2            25110000           170000                     490000   
3            24630000           190000                     510000   
4            24390000           210000                     490000   
...               ...              ...                        ...   
3883         76570000            30000                      40000   
3884         76890000            30000                     400000   
3885         75690000            40000                    1190000   
3886         75160000            30000                    1230000   
3887         76050000            30000                    1180000   

      monthly_insurance_loan  monthly_sbk_loan  loan_commitment  \
0                          0                 0          1120000   
1                     220000           1020000           680000   
2                     210000           1140000           640000   
3                     240000           1340000           510000   
4                     340000           1490000           640000   
...                      ...               ...              ...   
3883                  840000            450000          7290000   
3884                  840000            430000          7110000   
3885                  790000            450000          6950000   
3886                  860000            580000          6850000   
3887                  930000            470000          6480000   

      inst_rep_loanb  ls_rep_loanb  credit_loan  mortgage_loan  \
0            8210000      78020000     92170000       87260000   
1            4660000      23300000      8260000       42410000   
2            4370000      22930000      8300000       41710000   
3            4770000      22160000      9460000       41040000   
4            4740000      21960000      9660000       40850000   
...              ...           ...          ...            ...   
3883        21340000      57900000 

In [ ]:
df1.columns

Index(['pop_cd', 'year', 'month', 'avg_score', 'avg_rat', 'city', 'sex',
       'ages', 'population', 'num_opencard', 'num_usecard',
       'monthly_card_spend', 'monthly_lc', 'monthly_loan', 'monthly_bk_loan',
       'monthly_cd_loan', 'monthly_installments_loan',
       'monthly_insurance_loan', 'monthly_sbk_loan', 'loan_commitment',
       'inst_rep_loanb', 'ls_rep_loanb', 'credit_loan', 'mortgage_loan',
       'credit_card_payment', 'credit_card_installments_payment'],
      dtype='object')

In [ ]:
df1.duplicated().sum()

0

* 중복값 없음

In [ ]:
df1.isnull().sum()

pop_cd                                 0
year                                   0
month                                  0
avg_score                              0
avg_rat                                0
city                                 432
sex                                 3456
ages                                   0
population                             0
num_opencard                           0
num_usecard                            0
monthly_card_spend                     0
monthly_lc                             0
monthly_loan                           0
monthly_bk_loan                        0
monthly_cd_loan                        0
monthly_installments_loan              0
monthly_insurance_loan                 0
monthly_sbk_loan                       0
loan_commitment                        0
inst_rep_loanb                         0
ls_rep_loanb                           0
credit_loan                            0
mortgage_loan                          0
credit_card_paym

* 성별(sex) 정보는 못쓸 듯 

In [ ]:
df1.city.value_counts()

서울    216
인천    216
대전    216
대구    216
광주    216
부산    216
울산    216
경기    216
강원    216
충북    216
충남    216
전북    216
전남    216
경북    216
경남    216
제주    216
Name: city, dtype: int64

In [ ]:
df1[df1.city.isnull()]

pop_cd  year  month  avg_score  avg_rat city sex ages  population  \
384     Y10  2016      1        811        2  NaN  남자  10대      960000   
385     Y10  2016      2        767        4  NaN  남자  10대     1144000   
386     Y10  2016      3        765        4  NaN  남자  10대     1168000   
387     Y10  2016      4        764        4  NaN  남자  10대     1211000   
388     Y10  2016      5        765        4  NaN  남자  10대     1216000   
...     ...   ...    ...        ...      ...  ...  ..  ...         ...   
3883    Y95  2017      8        819        3  NaN  여자  90대      232000   
3884    Y95  2017      9        819        3  NaN  여자  90대      235000   
3885    Y95  2017     10        819        3  NaN  여자  90대      239000   
3886    Y95  2017     11        819        3  NaN  여자  90대      243000   
3887    Y95  2017     12        820        3  NaN  여자  90대      265000   

      num_opencard  num_usecard  monthly_card_spend  monthly_lc  monthly_loan  \
384           1.16         1.05               50000    53760000      48700000   
385           1.59         0.90              150000    19520000      16360000   
386           1.62         0.90              170000    19050000      16400000   
387           1.66         0.89              180000    19400000      16880000   
388           1.66         0.91              150000    19570000      16380000   
...            ...          ...                 ...         ...           ...   
3883          0.11         0.03               90000   100520000      86580000   
3884          0.11         0.03               90000   100580000      86780000   
3885          0.11         0.03               90000   100380000      86110000   
3886          0.11         0.03               90000   100370000      85920000   
3887          0.11         0.03               90000   101890000      86660000   

      monthly_bk_loan  monthly_cd_loan  monthly_installments_loan  \
384          46120000                0                      40000   
385          12090000           180000                    1470000   
386          11940000           180000                    1490000   
387          12050000           190000                    1520000   
388          12050000           200000                    1650000   
...               ...              ...                        ...   
3883         76570000            30000                      40000   
3884         76890000            30000                     400000   
3885         75690000            40000                    1190000   
3886         75160000            30000                    1230000   
3887         76050000            30000                    1180000   

      monthly_insurance_loan  monthly_sbk_loan  loan_commitment  \
384                        0                 0           940000   
385                   400000            300000          1160000   
386                   470000            260000          1100000   
387                   670000            220000          1070000   
388                   690000            220000           580000   
...                      ...               ...              ...   
3883                  840000            450000          7290000   
3884                  840000            430000          7110000   
3885                  790000            450000          6950000   
3886                  860000            580000          6850000   
3887                  930000            470000          6480000   

      inst_rep_loanb  ls_rep_loanb  credit_loan  mortgage_loan  \
384          2310000      45440000     16760000       49540000   
385          5560000       9340000      7830000       19930000   
386          5860000       9160000      7750000       20090000   
387          6250000       9260000      8310000       20620000   
388          6360000       9140000      8630000       19660000   
...              ...           ...          ...            ...   
3883        21340000      57900000     26130000

In [ ]:
df1.pop_cd.apply(lambda x: x[:1]).value_counts()

L    3456
Y     432
Name: pop_cd, dtype: int64

In [ ]:
df_city_null = df1[df1.city.isnull()]
df_city_null.pop_cd.apply(lambda x: x[:1]).value_counts()

Y    432
Name: pop_cd, dtype: int64

* 고객구분코드(pop_cd) ~~알파벳의 정확한 의미는 모르겠으나~~ Y 고객의 경우 지역(city)값이 결측
* ~~모르는 지역이 두 개 정도 합쳐진 느낌인데 (city 값 216개씩) 데이터를 모르겠어서 __빼고 진행할지 논의 필요~~__
* 데이콘 답변: __L 시리즈는 연령X지역 조합입니다. Y 시리즈는 연령대별 통계를 남자|여자 구분되어 있습니다__  -> 결국 같은 자료를 가지고 조합을 다르게 한거라 데이터프레임을 분리해서 봐야될듯

In [ ]:
df_city_null.sex.isnull().sum()

0

* 추가로 지역이 결측인 대신 성별 정보는 다 가지고 있는 것을 확인... 뭐지?!
* 암튼 __성별이 결측이면 지역을 알 수 있고, 성별을 알 수 있으면 지역이 결측인 상황__

In [ ]:
df1.groupby(['year', 'month']).count()

pop_cd  avg_score  avg_rat  city  sex  ages  population  \
year month                                                            
2016 1         162        162      162   144   18   162         162   
     2         162        162      162   144   18   162         162   
     3         162        162      162   144   18   162         162   
     4         162        162      162   144   18   162         162   
     5         162        162      162   144   18   162         162   
     6         162        162      162   144   18   162         162   
     7         162        162      162   144   18   162         162   
     8         162        162      162   144   18   162         162   
     9         162        162      162   144   18   162         162   
     10        162        162      162   144   18   162         162   
     11        162        162      162   144   18   162         162   
     12        162        162      162   144   18   162         162   
2017 1         162        162      162   144   18   162         162   
     2         162        162      162   144   18   162         162   
     3         162        162      162   144   18   162         162   
     4         162        162      162   144   18   162         162   
     5         162        162      162   144   18   162         162   
     6         162        162      162   144   18   162         162   
     7         162        162      162   144   18   162         162   
     8         162        162      162   144   18   162         162   
     9         162        162      162   144   18   162         162   
     10        162        162      162   144   18   162         162   
     11        162        162      162   144   18   162         162   
     12        162        162      162   144   18   162         162   

            num_opencard  num_usecard  monthly_card_spend  monthly_lc  \
year month                                                              
2016 1               162          162                 162         162   
     2               162          162                 162         162   
     3               162          162                 162         162   
     4               162          162                 162         162   
     5               162          162                 162         162   
     6               162          162                 162         162   
     7               162          162                 162         162   
     8               162          162                 162         162   
     9               162          162                 162         162   
     10              162          162                 162         162   
     11              162          162                 162         162   
     12              162          162                 162         162   
2017 1               162          162                 162         162   
     2               162          162                 162         162   
     3               162          162                 162         162   
     4               162          162                 162         162   
     5               162          162                 162         162   
     6               162          162                 162         162   
     7               162          162                 162         162   
     8               162          162                 162         162   
     9               162          162                 162         162   
     10              162          162                 162         162   
     11              162          162                 162         162   
     12              162          162                 162         162   

            monthly_loan  monthly_bk_loan  monthly_cd_loan  \
year month                                                   
2016 1               162              162              162   
     2               162              162              162   
     3            

In [ ]:
df1.pop_cd.nunique()

162

In [ ]:
print(df1.shape[0])
print(162*12*2)

3888
3888


* 2016.01-2017.12 기간의 자료
* 고객구분은 162개로 나눈 듯함  
* 162(그룹) * 12(개월) * 2(년)=3,888개의 데이터
* 162 = 144(지역)+18(성별)
* 데이콘 답변: 고객구분 그룹은 데이터 필드에 명세된 코드 값과 등록년도-등록월에 해당하는 그룹입

In [ ]:
df_a = df1[~df1.city.isnull()].dropna(axis=1)
df_b = df_city_null.dropna(axis=1)
display(df_a)
display(df_b)

pop_cd  year  month  avg_score  avg_rat city ages  population  \
0      L011  2016      1        812        2   서울  10대       55000   
1      L011  2016      2        773        4   서울  10대      380000   
2      L011  2016      3        771        4   서울  10대      387000   
3      L011  2016      4        770        4   서울  10대      412000   
4      L011  2016      5        771        4   서울  10대      414000   
...     ...   ...    ...        ...      ...  ...  ...         ...   
3835   L169  2017      8        810        3   제주  90대        2000   
3836   L169  2017      9        809        3   제주  90대        2000   
3837   L169  2017     10        809        3   제주  90대        2000   
3838   L169  2017     11        810        3   제주  90대        2000   
3839   L169  2017     12        811        3   제주  90대        2000   

      num_opencard  num_usecard  monthly_card_spend  monthly_lc  monthly_loan  \
0             1.44         1.13               70000    92630000      87350000   
1             1.71         0.96              170000    31140000      28910000   
2             1.75         0.95              190000    30230000      28180000   
3             1.76         0.92              200000    30430000      27710000   
4             1.75         0.94              170000    31210000      27620000   
...            ...          ...                 ...         ...           ...   
3835          0.38         0.07               50000    36320000      28950000   
3836          0.41         0.06               30000    36470000      29440000   
3837          0.42         0.06               50000    34190000      26850000   
3838          0.43         0.05               40000    48420000      41250000   
3839          0.43         0.06               50000    48620000      41430000   

      monthly_bk_loan  monthly_cd_loan  monthly_installments_loan  \
0            85790000                0                          0   
1            25850000           190000                     510000   
2            25110000           170000                     490000   
3            24630000           190000                     510000   
4            24390000           210000                     490000   
...               ...              ...                        ...   
3835         19020000                0                          0   
3836         20080000                0                          0   
3837         17240000                0                          0   
3838         32170000                0                          0   
3839         32110000                0                          0   

      monthly_insurance_loan  monthly_sbk_loan  loan_commitment  \
0                          0                 0          1120000   
1                     220000           1020000           680000   
2                     210000           1140000           640000   
3                     240000           1340000           510000   
4                     340000           1490000           640000   
...                      ...               ...              ...   
3835                       0                 0          3000000   
3836                       0                 0          2830000   
3837                       0                 0          2940000   
3838                       0                 0          2780000   
3839                       0                 0          2830000   

      inst_rep_loanb  ls_rep_loanb  credit_loan  mortgage_loan  \
0            8210000      78020000     92170000       87260000   
1            4660000      23300000      8260000       42410000   
2            4370000      22930000      8300000       41710000   
3            4770000      22160000      9460000       41040000   
4            4740000      21960000      9660000       40850000   
...              ...           ...          ...            ...   
3835         6990000      18960000     17060000       35400000   
3836         7640000      189

pop_cd  year  month  avg_score  avg_rat sex ages  population  \
384     Y10  2016      1        811        2  남자  10대      960000   
385     Y10  2016      2        767        4  남자  10대     1144000   
386     Y10  2016      3        765        4  남자  10대     1168000   
387     Y10  2016      4        764        4  남자  10대     1211000   
388     Y10  2016      5        765        4  남자  10대     1216000   
...     ...   ...    ...        ...      ...  ..  ...         ...   
3883    Y95  2017      8        819        3  여자  90대      232000   
3884    Y95  2017      9        819        3  여자  90대      235000   
3885    Y95  2017     10        819        3  여자  90대      239000   
3886    Y95  2017     11        819        3  여자  90대      243000   
3887    Y95  2017     12        820        3  여자  90대      265000   

      num_opencard  num_usecard  monthly_card_spend  monthly_lc  monthly_loan  \
384           1.16         1.05               50000    53760000      48700000   
385           1.59         0.90              150000    19520000      16360000   
386           1.62         0.90              170000    19050000      16400000   
387           1.66         0.89              180000    19400000      16880000   
388           1.66         0.91              150000    19570000      16380000   
...            ...          ...                 ...         ...           ...   
3883          0.11         0.03               90000   100520000      86580000   
3884          0.11         0.03               90000   100580000      86780000   
3885          0.11         0.03               90000   100380000      86110000   
3886          0.11         0.03               90000   100370000      85920000   
3887          0.11         0.03               90000   101890000      86660000   

      monthly_bk_loan  monthly_cd_loan  monthly_installments_loan  \
384          46120000                0                      40000   
385          12090000           180000                    1470000   
386          11940000           180000                    1490000   
387          12050000           190000                    1520000   
388          12050000           200000                    1650000   
...               ...              ...                        ...   
3883         76570000            30000                      40000   
3884         76890000            30000                     400000   
3885         75690000            40000                    1190000   
3886         75160000            30000                    1230000   
3887         76050000            30000                    1180000   

      monthly_insurance_loan  monthly_sbk_loan  loan_commitment  \
384                        0                 0           940000   
385                   400000            300000          1160000   
386                   470000            260000          1100000   
387                   670000            220000          1070000   
388                   690000            220000           580000   
...                      ...               ...              ...   
3883                  840000            450000          7290000   
3884                  840000            430000          7110000   
3885                  790000            450000          6950000   
3886                  860000            580000          6850000   
3887                  930000            470000          6480000   

      inst_rep_loanb  ls_rep_loanb  credit_loan  mortgage_loan  \
384          2310000      45440000     16760000       49540000   
385          5560000       9340000      7830000       19930000   
386          5860000       9160000      7750000       20090000   
387          6250000       9260000      8310000       20620000   
388          6360000       9140000      8630000       19660000   
...              ...           ...          ...            ...   
3883        21340000      57900000     26130000       98590000   
3884        21710000      57910000     30

In [ ]:
# df_a.to_csv('city_card_data_city.csv', encoding='utf-8-sig', index=False)
# df_b.to_csv('city_card_data_sex.csv', encoding='utf-8-sig', index=False)

* df_a: 연령X지역 통계
* df_b: 연령X성별 통계

In [ ]:
df_a.describe()

year        month    avg_score      avg_rat    population  \
count  3456.000000  3456.000000  3456.000000  3456.000000  3.456000e+03   
mean   2016.500000     6.500000   801.018808     3.244792  3.155009e+05   
std       0.500072     3.452552    22.849623     0.441975  4.573174e+05   
min    2016.000000     1.000000   726.000000     2.000000  2.000000e+03   
25%    2016.000000     3.750000   782.000000     3.000000  7.200000e+04   
50%    2016.500000     6.500000   807.000000     3.000000  2.000000e+05   
75%    2017.000000     9.250000   818.000000     3.250000  3.365000e+05   
max    2017.000000    12.000000   850.000000     4.000000  2.506000e+06   

       num_opencard  num_usecard  monthly_card_spend    monthly_lc  \
count   3456.000000  3456.000000        3.456000e+03  3.456000e+03   
mean       2.346670     1.286224        8.199740e+05  6.920643e+07   
std        1.389565     0.761683        6.207638e+05  3.151079e+07   
min        0.140000     0.030000        2.000000e+04  3.500000e+06   
25%        1.120000     0.630000        2.100000e+05  4.350000e+07   
50%        2.290000     1.340000        7.500000e+05  7.409500e+07   
75%        3.660000     1.950000        1.440000e+06  9.046500e+07   
max        5.170000     2.790000        2.300000e+06  1.817200e+08   

       monthly_loan  monthly_bk_loan  monthly_cd_loan  \
count  3.456000e+03     3.456000e+03     3.456000e+03   
mean   5.918240e+07     4.461245e+07     6.836429e+05   
std    2.720454e+07     2.333570e+07     5.490890e+05   
min    3.500000e+06     1.210000e+06     0.000000e+00   
25%    3.680500e+07     3.026750e+07     1.400000e+05   
50%    6.334500e+07     4.688000e+07     6.400000e+05   
75%    7.718250e+07     5.753500e+07     1.190000e+06   
max    1.658500e+08     1.506500e+08     2.040000e+06   

       monthly_installments_loan  monthly_insurance_loan  monthly_sbk_loan  \
count               3.456000e+03            3.456000e+03      3.456000e+03   
mean                2.420758e+06            1.640911e+06      8.481655e+05   
std                 1.791869e+06            1.425775e+06      7.679104e+05   
min                 0.000000e+00            0.000000e+00      0.000000e+00   
25%                 7.100000e+05            4.900000e+05      1.900000e+05   
50%                 2.290000e+06            1.185000e+06      6.500000e+05   
75%                 4.120000e+06            2.610000e+06      1.360000e+06   
max                 5.960000e+06            6.980000e+06      4.350000e+06   

       loan_commitment  inst_rep_loanb  ls_rep_loanb   credit_loan  \
count     3.456000e+03    3.456000e+03  3.456000e+03  3.456000e+03   
mean      1.105053e+07    1.845000e+07  2.869549e+07  1.979782e+07   
std       6.337683e+06    1.104747e+07  1.572765e+07  8.920733e+06   
min       0.000000e+00    0.000000e+00  2.020000e+06  0.000000e+00   
25%       5.972500e+06    9.150000e+06  1.852000e+07  1.301000e+07   
50%       1.135500e+07    1.573500e+07  2.863500e+07  1.840500e+07   
75%       1.579000e+07    2.580250e+07  3.679750e+07  2.694250e+07   
max       2.800000e+07    4.891000e+07  1.106000e+08  9.217000e+07   

       mortgage_loan  credit_card_payment  credit_card_installments_payment  
count   3.456000e+03         3.456000e+03                       3456.000000  
mean    7.593950e+07         4.172535e+05                     141024.016204  
std     3.494024e+07         3.712155e+05                     115669.031070  
min     3.500000e+06         0.000000e+00                          0.000000  
25%     4.794500e+07         6.400000e+04                      25000.000000  
50%     7.850000e+07         2.755000e+05                     101000.000000  
75%     9.911250e+07         7.822500e+05                     253000.000000  
max     1.830500e+08         1.367000e+06                     402000.000000

In [ ]:
df_b.describe()

year       month   avg_score     avg_rat    population  \
count   432.00000  432.000000  432.000000  432.000000  4.320000e+02   
mean   2016.50000    6.500000  803.525463    3.263889  2.766222e+06   
std       0.50058    3.456055   23.422762    0.451645  1.717282e+06   
min    2016.00000    1.000000  733.000000    2.000000  1.220000e+05   
25%    2016.00000    3.750000  783.750000    3.000000  1.125750e+06   
50%    2016.50000    6.500000  814.000000    3.000000  3.032000e+06   
75%    2017.00000    9.250000  822.000000    4.000000  4.530250e+06   
max    2017.00000   12.000000  839.000000    4.000000  5.160000e+06   

       num_opencard  num_usecard  monthly_card_spend    monthly_lc  \
count    432.000000   432.000000        4.320000e+02  4.320000e+02   
mean       2.221806     1.225370        8.494907e+05  7.684146e+07   
std        1.381078     0.755131        6.204943e+05  2.917882e+07   
min        0.090000     0.020000        5.000000e+04  1.277000e+07   
25%        1.077500     0.567500        2.000000e+05  6.761500e+07   
50%        2.150000     1.260000        7.750000e+05  8.592000e+07   
75%        3.540000     1.890000        1.482500e+06  9.940500e+07   
max        4.650000     2.620000        2.160000e+06  1.153200e+08   

       monthly_loan  monthly_bk_loan  monthly_cd_loan  \
count  4.320000e+02     4.320000e+02     4.320000e+02   
mean   6.599192e+07     5.255556e+07     7.569907e+05   
std    2.517319e+07     2.205274e+07     5.881827e+05   
min    1.190000e+07     6.760000e+06     0.000000e+00   
25%    5.670250e+07     4.195250e+07     1.600000e+05   
50%    7.378000e+07     5.820500e+07     7.400000e+05   
75%    8.612750e+07     7.179750e+07     1.270000e+06   
max    1.000100e+08     7.973000e+07     1.980000e+06   

       monthly_installments_loan  monthly_insurance_loan  monthly_sbk_loan  \
count               4.320000e+02            4.320000e+02      4.320000e+02   
mean                2.249954e+06            1.942894e+06      8.439120e+05   
std                 1.610679e+06            1.278306e+06      8.276140e+05   
min                 4.000000e+04            0.000000e+00      0.000000e+00   
25%                 6.875000e+05            8.700000e+05      2.775000e+05   
50%                 2.135000e+06            1.700000e+06      5.150000e+05   
75%                 3.692500e+06            2.892500e+06      1.245000e+06   
max                 5.450000e+06            4.720000e+06      4.290000e+06   

       loan_commitment  inst_rep_loanb  ls_rep_loanb   credit_loan  \
count     4.320000e+02    4.320000e+02  4.320000e+02  4.320000e+02   
mean      1.049741e+07    2.271958e+07  3.168727e+07  2.104597e+07   
std       6.012025e+06    1.009507e+07  1.444329e+07  8.251382e+06   
min       4.900000e+05    2.310000e+06  5.890000e+06  7.750000e+06   
25%       7.037500e+06    1.589250e+07  2.103000e+07  1.514000e+07   
50%       1.004000e+07    2.114500e+07  3.324500e+07  1.982000e+07   
75%       1.581500e+07    3.177000e+07  4.094000e+07  2.442250e+07   
max       2.086000e+07    4.436000e+07  5.927000e+07  3.999000e+07   

       mortgage_loan  credit_card_payment  credit_card_installments_payment  
count   4.320000e+02         4.320000e+02                        432.000000  
mean    8.576373e+07         4.416505e+05                     144988.425926  
std     3.142221e+07         3.823014e+05                     113705.209097  
min     1.292000e+07         0.000000e+00                          0.000000  
25%     8.093250e+07         1.017500e+05                      40750.000000  
50%     9.877000e+07         3.080000e+05                     117000.000000  
75%     1.061775e+08         8.150000e+05                     253250.000000  
max     1.261000e+08         1.326000e+06                     380000.000000

In [ ]:
!pip install dataprep
from dataprep.eda import create_report

     |████████████████████████████████| 9.5 MB 1.0 MB/s 
     |████████████████████████████████| 1.0 MB 49.6 MB/s 
     |████████████████████████████████| 95 kB 3.1 MB/s 
     |████████████████████████████████| 133 kB 43.5 MB/s 
     |████████████████████████████████| 1.5 MB 47.2 MB/s 
     |████████████████████████████████| 943 kB 41.1 MB/s 
     |████████████████████████████████| 1.1 MB 17.2 MB/s 
     |████████████████████████████████| 749 kB 32.7 MB/s 
     |████████████████████████████████| 78 kB 3.9 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 28.5 MB 60.9 MB/s 
     |████████████████████████████████| 10.9 MB 14.7 MB/s 
     |████████████████████████████████| 366 kB 30.6 MB/s 
     |████████████████████████████████| 50 kB 5.0 MB/s 
     |████████████████████████████████| 965 kB 46.9 MB/s 
     |████████████████████████████████| 271 kB 58.1 MB/s 
     |███

In [ ]:
create_report(df_a)

Output hidden; open in https://colab.research.google.com to view.

In [ ]:
create_report(df_b)

Output hidden; open in https://colab.research.google.com to view.

# 2. jeju_financial_life_data.csv 
> __제주도__의 우편번호 단위 지역민의 __금융생활 통계 데이터__로 우편번호 단위 별, 거주지 블록 별 __소득 비교, 채무 현황__ 등 다양한 컬럼을 활용하여 인사이트를 발굴할 수 있다.

## __지역 거주지(제주도) 금융라이프 데이터__



### KEY Dimmension  
1. zip_cd 신우편번호 신우편번호  지역: 제주특별자치도
2. year_month 기준월 해당 정보 자료의 시점기준 정보   
3. x_axis 좌표 해당 우편번호의 경도   
4. y_axis 좌표 해당 우편번호의 위도   
5. sex 성별 1: 남성, 2: 여성   
6. age 연령대 5세 구간(24: 24세이하, 29: 25-29세, 34: 30-34세, …. , 74: 70-74세, 79: 75-79세, 99: 80세이상)   

### 직업관련
1. job_majorc 직업군별 비율-대기업 고객군의 직업군별 비율 : 대기업 ratio 범위 : 0~1
2. job_smallc 직업군별 비율-중소기업 고객군의 직업군별 비율 : 중소기업 ratio 범위 : 0~1
3. job_public 직업군별 비율-공기업 고객군의 직업군별 비율 : 공기업 ratio 범위 : 0~1
4. job_profession 직업군별 비율-전문직 고객군의 직업군별 비율 : 전문직 ratio 범위 : 0~1
5. job_self 직업군별 비율-자영업 고객군의 직업군별 비율 : 자영업 ratio 범위 : 0~1
6. job_none 직업군별 비율-무직 고객군의 직업군별 비율 : 무직 ratio 범위 : 0~1
7. job_other 직업분별 비율-기타 고객군의 직업군별 비율 : 판단불가 ratio 범위 : 0~1

### 소득관련
1. avg_income 평균연소득 고객군별 평균 KCB 결정연소득 numeric 단위 : 원
2. med_income 중위연소득 고객군별 KCB 결정연소득의 중위수(median)값 numeric 단위 : 원

### 소비관련
1. avg_spend 평균소비액 고객군별 3개월 평균 KCB 총카드이용금액 numeric 단위 : 원
2. avg_foreign_spend 평균해외소비액 고객군별 3개월 평균 KCB 총 카드해외이용금액 numeric 단위 : 원



### 부채관련
1. avg_debt 채무 평균보유액 고객군별 채무보유자의 평균 KCB 총대출잔액 numeric 단위 : 원
2. avg_debt_credit 채무 평균보유액-신용대출 고객군별 채무보유자의 평균 KCB 신용대출잔액 numeric 단위 : 원
4. avg_debt_noneb 채무 평균보유액-비은행대출 고객군별 채무보유자의 평균 KCB 비은행권대출잔액 numeric 단위 : 원
5. avg_debt_mortgage 채무 평균보유액-주택담보대출 고객군별 채무보유자의 평균 KCB 주택담보대출잔액 numeric 단위 : 원
6. avg_debt_deposit 채무 평균보유액-예적금유가증권담보대출 고객군별 채무보유자의 평균 KCB 예적금 또는 유가증권 담보대출 잔액 numeric 단위 : 원
7. avg_debt_collateral 채무 평균보유액-물건담보대출 고객군별 채무보유자의 평균 KCB 물건담보(주거용 및 비주거용 부동산, 자동차 등) 대출 잔액 numeric 단위 : 원

### 신용도판단정보 
1. avg_credit_rat 평균관리지수 고객군별 신용평균 numeric  

### 부동산관련
1. medium_resid_rat 중대형이상 거주비율 고객군의 중대형 평형 거주비율(30평 이상) (-999,999: 부동산가격이 없는경우) ratio 범위 : 0~1
2. large_resid_rat 대형 거주비율 고객군의 대형 평형 거주비율(40평 이상) (-999,999: 부동산가격이 없는경우) ratio 범위 : 0~1
3. vehicle_own_rat 중대형 차량소유비율 고객군별 중대형 차량소유비율 ratio 범위 : 0~1



In [ ]:
df2

zip_cd  year_month      x_axis     y_axis  sex  age  job_majorc  \
0       63299      201902  126.565889  33.513210    1   54    0.000000   
1       63560      201902  126.497121  33.247225    1   59    0.000000   
2       63560      201902  126.497121  33.247225    2   49    0.000000   
3       63560      201902  126.497121  33.247225    1   54    0.000000   
4       63560      201902  126.497121  33.247225    1   49    0.000000   
...       ...         ...         ...        ...  ...  ...         ...   
10417   63268      201902  126.531404  33.509626    1   39    0.017544   
10418   63343      201902  126.665089  33.501399    1   59    0.028571   
10419   63072      201902  126.426763  33.485506    2   44    0.029412   
10420   63343      201902  126.665089  33.501399    1   64    0.052632   
10421   63343      201902  126.665089  33.501399    2   39    0.081081   

       job_smallc  job_public  job_profession  ...   avg_debt  \
0        0.166667    0.000000        0.000000  ...   24285196   
1        0.125000    0.000000        0.000000  ...  108571077   
2        0.250000    0.000000        0.000000  ...   26424335   
3        0.181818    0.090909        0.090909  ...  647716901   
4        0.250000    0.000000        0.000000  ...  248232725   
...           ...         ...             ...  ...        ...   
10417    0.385965    0.035088        0.017544  ...   51576038   
10418    0.085714    0.085714        0.000000  ...  207666185   
10419    0.264706    0.000000        0.029412  ...   53861994   
10420    0.078947    0.052632        0.000000  ...  158266839   
10421    0.324324    0.054054        0.000000  ...   68784573   

       avg_debt_credit  avg_debt_noneb  avg_debt_mortgage  avg_debt_deposit  \
0             11419148         8863595           55550000                 0   
1             25896955        13317443                  0         200000000   
2             12015311        12015311           40833359                 0   
3             39952556        21558153           36697792                 0   
4             35712501         5568751                  0                 0   
...                ...             ...                ...               ...   
10417         23110274        13431142           89315216                 0   
10418         42386142        22437308          184612481          13600000   
10419         16967315        37701875           90772396            100000   
10420         34091265        28895365          186087107                 0   
10421         35872873        59023088          120188650                 0   

       avg_debt_collateral  avg_credit_rat  medium_resid_rat  large_resid_rat  \
0                 55550000               5         -999999.0        -999999.0   
1                 74421721               4         -999999.0        -999999.0   
2                 40833359               2         -999999.0        -999999.0   
3                706300496               4         -999999.0        -999999.0   
4                626810674               5         -999999.0        -999999.0   
...                    ...             ...               ...              ...   
10417            107706758               4         -999999.0        -999999.0   
10418            250278827               3         -999999.0        -999999.0   
10419            112076182               4         -999999.0        -999999.0   
10420            194598568               3         -999999.0        -999999.0   
10421            120188650               3         -999999.0        -999999.0   

       vehicle_own_rat  
0             0.000000  
1             0.000000  
2             0.000000  
3             0.166667  
4             0.000000  
...                ...  
10417         0.050000  
10418         0.052632  
10419         0.114286  
10420         0.048780  
10421         0.076923  

[10422 rows x 27 columns]

In [ ]:
df2.columns

Index(['zip_cd', 'year_month', 'x_axis', 'y_axis', 'sex', 'age', 'job_majorc',
       'job_smallc', 'job_public', 'job_profession', 'job_self', 'job_none',
       'job_other', 'avg_income', 'med_income', 'avg_spend',
       'avg_foreign_spend', 'avg_debt', 'avg_debt_credit', 'avg_debt_noneb',
       'avg_debt_mortgage', 'avg_debt_deposit', 'avg_debt_collateral',
       'avg_credit_rat', 'medium_resid_rat', 'large_resid_rat',
       'vehicle_own_rat'],
      dtype='object')

In [ ]:
df2.duplicated().sum()

0

In [ ]:
df2.isnull().sum()

zip_cd                 0
year_month             0
x_axis                 0
y_axis                 0
sex                    0
age                    0
job_majorc             0
job_smallc             0
job_public             0
job_profession         0
job_self               0
job_none               0
job_other              0
avg_income             0
med_income             0
avg_spend              0
avg_foreign_spend      0
avg_debt               0
avg_debt_credit        0
avg_debt_noneb         0
avg_debt_mortgage      0
avg_debt_deposit       0
avg_debt_collateral    0
avg_credit_rat         0
medium_resid_rat       0
large_resid_rat        0
vehicle_own_rat        0
dtype: int64

In [ ]:
df2.year_month.unique()

array([201902])

* 제주 데이터는 시계열 아님! 2019.02 기준 데이터

In [ ]:
create_report(df2)

Output hidden; open in https://colab.research.google.com to view.